src/NinjaTools.FluentMockServer.API/Configuration/ConfigFileProvider.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using NinjaTools.FluentMockServer.API.Models;
+using YamlDotNet.Serialization;
+
+namespace NinjaTools.FluentMockServer.API.Configuration
+{
+   public class ConfigFileProvider
+    {
+        private readonly IFileSystem _fs;
+        private readonly ILogger<ConfigFileProvider> _logger;
+
+        public ConfigFileProvider(IFileSystem fs, ILogger<ConfigFileProvider> logger)
+        {
+            _fs = fs;
+            _logger = logger;
+        }
+
+        private  ConfigFile ParseJson(string path)
+        {
+            var jo = JArray.Parse(_fs.File.ReadAllText(path));
+            var setups = jo.ToObject<List<Setup>>();
+
+            return new ConfigFile(path, setups.ToArray());
+        }
+
+        private  IConfigFile ParseYaml(string path)
+        {
+            using var reader = new StringReader(_fs.File.ReadAllText(path));
+            var deserializer = new DeserializerBuilder().Build();
+
+            var setups = deserializer.Deserialize<List<Setup>>(reader);
+            return new ConfigFile(path, setups.ToArray());
+        }
+
+        public IEnumerable<IConfigFile> GetConfigFiles(string rootDirectory)
+        {
+            var files = _fs.Directory.GetFiles(rootDirectory, "*", SearchOption.AllDirectories);
+
+            foreach (var file in files)
+            {
+                var fileExtension = _fs.Path.GetExtension(file).TrimStart('.').ToLower();
+
+                if (Enum.TryParse<ConfigurationFileType>(fileExtension, out var fileType))
+                {
+                    foreach (var configFile in ParseConfigFile(fileType, file))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'foreach' block.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4765163)